### ISSUE TITLE
[1.18.2] Chainsaw Mk1 drops nothing.

### ISSUE THREAD

**Issue by Yirok42** [mcenderdragon/Futurepack-API#865]
**Describe the bug**
Instead of felling an entire tree or entire tree's canopy, the chainsaw seems to break blocks similar to creative mode ... drops nothing.

**To Reproduce**
1. Build the Chainsaw Mk1
2. Charge it to 2500.
3. Use it on a tree or leaves.
4. Block disappears but drops nothing.

**Version**
Operating System: Windows 10 version 10.0
Minecraft Version: 1.18.2
Java Version: 17.0.2, Oracle Corporation
forge-1.18.2-40.0.4
Futurepack-1.18.2-33.0.7227
create-mc1.18.2_v0.4.1
flywheel-forge-1.18-0.6.2
corpse-1.18.2-1.0.1
jei-1.18.2-9.5.5.174
journeymap-1.18.2-5.8.0

**Comment by mcenderdragon** [comment]
did you make sure to charge the CHainsaw? I noticed, all the bars are wrong

**Comment by Yirok42** [comment]
I charged it back to 2500 and tried it on an oak tree.

The game crashed whilst ticking entity
Error: java.lang.ArrayStoreException: net.minecraft.core.Holder$Reference
Exit Code: -1

I generally ignore these ticking entity crashes but since I already have a ticket going I thought I would share.

https://pastebin.com/2UqCHr1z

**Comment by Yirok42** [comment]
Second try, I made sure it was at 2500 of 2500, and I cut a log from an oak tree (same tree as the crash), and the block broke and the chainsaw lost no energy. So I broke a leaf block of the same tree, and the chainsaw dropped to 2490 of 2500. So I broke another log block and it did not decrease. I then broke another leaf block and I got a ticking entity crash so I am putting the chainsaw back into the wardrobe to sit.
